frontend/apps/reader/modules/readerannotation.lua

```diff
@@ -11,6 +11,12 @@ local ReaderAnnotation = WidgetContainer:extend{
 
 function ReaderAnnotation:buildAnnotation(bm, highlights, init)
     -- bm: associated single bookmark ; highlights: tables with all highlights
+    if self.ui.rolling and
+            (not self.document:isXPointerInDocument(bm.page) or
+            (bm.pos1 and not self.document:isXPointerInDocument(bm.pos1))) then
+        logger.warn("Invalid xpointer in old highlight, skipping")
```

Comment:
Might be worth logging the xpointer too, so users can find it out and see what text it is about, and possibly fix them manually (or rehighlight them)?